packages/analytics-js-plugins/src/beaconQueue/Queue.ts

```diff
@@ -0,0 +1,172 @@
+import Emitter from 'component-emitter';
+import { IStoreManager, StorageType, IStore, BeaconQueueOpts } from '../types/common';
+import { generateUUID } from '../utilities/common';
+import { IQueue, QueueItem, QueueProcessCallback } from '../types/plugins';
+import { BeaconQueueItem } from './types';
+import { getDeliveryPayload } from './utilities';
+import { DEFAULT_BEACON_QUEUE_OPTIONS, MAX_BATCH_PAYLOAD_SIZE_BYTES } from './constants';
+
+export type BeaconQueueTimeouts = {
+  flushQueueTimeOutInterval: number;
+};
+
+const sortByTime = (a: QueueItem, b: QueueItem) => a.time - b.time;
+
+class Queue extends Emitter implements IQueue<BeaconQueueItem> {
```

Comment:
Can we get rid of Emitter? Even previously, this was not there.
I believe we can remove it from the retry queue as well.

Reply:
Need to ask product, because Segemnt offer the feature (https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#emitter) and client may want to "hook" on processed data.

will start a discussion

Reply:
we drop the emitter, will remove it, done